fix(mermaid): zoom and reset diagram relative to the browser window zoom

vazome opened the pull request:
As per discord discussion: [Issues with mermaid diagrams](https://discord.com/channels/927628110009098281/1385134733192855572)

Fixes issue N2: mermaid center/zoom offset.

Before

https://github.com/user-attachments/assets/dac11518-ae34-42f5-b65b-2b3283593593

After

https://github.com/user-attachments/assets/c4d72bec-966c-4b7c-b915-7b251f4f7ae6

vazome committed:
fix(mermaid): zoom and reset digram relative to the browser window zoom

vazome committed:
prettier

saberzero1 commented:
Thanks for this. It seems the reset button doesn't properly account for scale/zoom level. See:

https://github.com/user-attachments/assets/161fea14-f423-44de-9dd0-e9682b2222b9

vazome committed:
mermaid-controls adjust

vazome committed:
mermaid-controls adjust

vazome committed:
mermaid-controls adjust

vazome commented:
Addressed both mermaid reset and browser zoom via `resize` event listener.

I've also fixed .mermaid-controls block: it was scaling with browser zoom, now it respects the window size.

https://github.com/user-attachments/assets/99abeb97-2acb-4357-b862-99b93956361d

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -86,30 +86,30 @@ pre {
       display: flex;
-      gap: 8px;
-      padding: 8px;
+      gap: 0.5vw;
```
Assume your scale is `8px` = `1vh`/`1vw`, shouldn't this be `1vw`?

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -86,30 +86,30 @@ pre {
-      gap: 8px;
-      padding: 8px;
+      gap: 0.5vw;
+      padding: 1vh 0.5vw;
```
`1vh 1vw`

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -86,30 +86,30 @@ pre {
       background: var(--light);
-      border: 1px solid var(--lightgray);
-      border-radius: 6px;
+      border: 0.2vh solid var(--lightgray);
```
`0.125vh`

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -86,30 +86,30 @@ pre {
-      border: 1px solid var(--lightgray);
-      border-radius: 6px;
+      border: 0.2vh solid var(--lightgray);
+      border-radius: 0.8vh;
```
`0.75vh`

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -86,30 +86,30 @@ pre {
         justify-content: center;
-        width: 32px;
-        height: 32px;
+        width: 4.5vh;
```
`4vw` (also not `vh`(?))

vazome replied:
Yeah, this one should be 2.5 vw with existing config

vazome replied:
```suggestion
        width: 2.5vw;
```

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -86,30 +86,30 @@ pre {
-        width: 32px;
-        height: 32px;
+        width: 4.5vh;
+        height: 4.5vh;
```
`4vh`

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -86,30 +86,30 @@ pre {
+        height: 4.5vh;
         padding: 0;
-        border: 1px solid var(--lightgray);
+        border: 0.2vh solid var(--lightgray);
```
`0.125vh`

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -121,11 +121,10 @@ pre {
           width: auto;
-          padding: 0 12px;
-          font-size: 14px;
+          padding: 0 1vw;
```
`1.5vw`

saberzero1 commented on quartz/components/styles/mermaid.inline.scss:
```diff
@@ -121,11 +121,10 @@ pre {
-          padding: 0 12px;
-          font-size: 14px;
+          padding: 0 1vw;
+          font-size: 2vh;
```
`1.75vh`

saberzero1 reviewed (commented):
Thanks for the changes.

I have a few small remarks related to the various `vh`/`vw` scales. It seems you use the scales inconsistently, and one width seems to use a `vh` value instead of `vw`.

I have left some remarks. These remarks were made with the assumption that `8px` = `1vh`/`1vw`.

vazome commented:
@saberzero1 regarding the scale, I was trying to make the mermaid control button to remain its size; with proposed values it looks stretched:

Proposed
<img alt="image" src="https://github.com/user-attachments/assets/7128fec1-2f29-438b-830b-57c87ea150d8" width="300">

Current
<img alt="image" src="https://github.com/user-attachments/assets/086d2184-2ded-48c0-aa5a-e7167fc49ba3" width="300">


`px` varies depending on the device, so I wouldn't asume it is `8px` = `1vh`/`vw`.
